File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/php/example_code/kinesisfirehose/CreateDeliveryStream.php
<?php
/**
 * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * This file is licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of
 * the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 *  ABOUT THIS PHP SAMPLE: This sample is part of the AWS SDK for PHP Developer Guide topic at 
 *  https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/kinesis-firehose-example-delivery-stream.html
 */

require 'vendor/autoload.php';

use Aws\Firehose\FirehoseClient; 
use Aws\Exception\AwsException;

/**
 * Creating an Amazon Kinesis Firehose Delivery Stream.
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

//Create a KinesisClient 
$firehoseClient = new Aws\Firehose\FirehoseClient([
    'profile' => 'default',
    'version' => '2015-08-04',
    'region' => 'us-east-2'
]);

$name = "my_stream_name";
$stream_type = "KinesisStreamAsSource";
$kinesis_stream = "arn:aws:kinesis:us-east-2:0123456789:stream/my_stream_name";
$role = "arn:aws:iam::0123456789:policy/Role";

try {
    $result = $firehoseClient->createDeliveryStream([
        'DeliveryStreamName' => $name,
        'DeliveryStreamType' => $stream_type,
        'KinesisStreamSourceConfiguration' => [
            'KinesisStreamARN' => $kinesis_stream,
            'RoleARN' => $role,
        ],
    ]);
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    echo $e->getMessage();
    echo "\n";
}
 
 

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/firehose/src/main/kotlin/com/kotlin/firehose/CreateDeliveryStream.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.firehose

import aws.sdk.kotlin.services.firehose.FirehoseClient
import aws.sdk.kotlin.services.firehose.model.ExtendedS3DestinationConfiguration
import aws.sdk.kotlin.services.firehose.model.CreateDeliveryStreamRequest
import aws.sdk.kotlin.services.firehose.model.DeliveryStreamType
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args:Array<String>) {

    val usage = """
    
    Usage:
        <bucketARN> <roleARN> <streamName> 

    Where:
        bucketARN - the ARN of the Amazon S3 bucket where the data stream is written. 
        roleARN - the ARN of the IAM role that has the permissions that Kinesis Data Firehose needs. 
        streamName - the name of the delivery stream. 
    """

    if (args.size != 3) {
        println(usage)
        exitProcess(0)
    }

    val bucketARN = args[0]
    val roleARN = args[1]
    val streamName = args[2]
    createStream(bucketARN, roleARN, streamName)
}

suspend fun createStream(bucketARNVal: String?, roleARNVal: String?, streamName: String?) {

        val destinationConfiguration = ExtendedS3DestinationConfiguration {
            bucketArn = bucketARNVal
            roleArn = roleARNVal
        }

       val request = CreateDeliveryStreamRequest {
           deliveryStreamName = streamName
           extendedS3DestinationConfiguration = destinationConfiguration
           deliveryStreamType = DeliveryStreamType.DirectPut
       }

       FirehoseClient { region = "us-west-2" }.use { firehoseClient ->
        val streamResponse = firehoseClient.createDeliveryStream(request)
        println("Delivery Stream ARN is ${streamResponse.deliveryStreamArn}")
       }
}
